pinot-controller/src/main/java/org/apache/pinot/controller/api/debug/TableDebugInfo.java

```diff
@@ -102,21 +113,71 @@ public int getNumBrokers() {
     return _brokerDebugInfos;
   }
 
+  @JsonPropertyOrder({"segmentName", "serverState"})
   public static class SegmentDebugInfo {
     private final String _segmentName;
-    private final List<IsEvState> _states;
 
-    public SegmentDebugInfo(String name, List<IsEvState> states) {
-      _segmentName = name;
-      _states = states;
+    private final Map<String, SegmentState> _serverStateMap;
+
+    @JsonCreator
+    public SegmentDebugInfo(@JsonProperty("segmentName") String segmentName,
+        @JsonProperty("serverState") Map<String, SegmentState> segmentServerState) {
+      _segmentName = segmentName;
+      _serverStateMap = segmentServerState;
     }
 
     public String getSegmentName() {
       return _segmentName;
     }
 
-    public List<IsEvState> getSegmentStateInServer() {
-      return _states;
+    public Map<String, SegmentState> getServerState() {
+      return _serverStateMap;
+    }
+  }
+
+  /**
+   * This class represents the state of segment on the server:
+   *
+   * <ul>
+   *   <li>Ideal State vs External view.</li>
+   *   <li>Segment related errors and consumer information.</li>
+   *   <li>Segment size.</li>
+   * </ul>
+   */
+  public static class SegmentState {
+    private final String _isState;
+    private final String _evState;
+    private final String _segmentSize;
+    private final SegmentConsumerInfo _consumerInfo;
+    private final SegmentErrorInfo _errorInfo;
```

Comment:
So, SegmentErrorInfo is propagated as it is to external world?

Reply:
Yes. Not sure if I follow the concern.

Reply:
The concern here is that it is also used in the server-controller protocol. So, it may come to play during upgrades, esp if there are automated systems issuing monitoring commands.

Reply:
We will have to maintain the controller-server protocol regardless, and ensure that upgrades are not broken. The debug endpoints are meant for human debugging, and not to build any systems around them.

Specifically for SegmentErrorInfo, I was in double minds about having two separate classes, one for server side and another for controller side, but wasn't convinced. One question I have is would JSON annotations on the class be honored by rest, if so, it should be equivalent to passing JSON between server-controller?

Thoughts?

Reply:
Merging PR, we can followup in subsequent if there are further concerns.